From c656183adf03fb5557d0bac5d3a62485252afb28 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 17 Aug 2016 17:52:58 +0200
Subject: [PATCH] make consolefont service compatible with busyboxs setfont
 applet

Compared to kdbs setfont program it doesn't support -O and -m.
---
 conf.d/consolefont    | 11 ++---------
 init.d/consolefont.in |  7 ++-----
 2 files changed, 4 insertions(+), 14 deletions(-)

diff --git a/conf.d/consolefont b/conf.d/consolefont
index e01ae842..75544b2f 100644
--- a/conf.d/consolefont
+++ b/conf.d/consolefont
@@ -3,16 +3,9 @@
 #
 # consolefont specifies the default font that you'd like Linux to use on the
 # console.  You can find a good selection of fonts in /usr/share/consolefonts;
-# you shouldn't specify the trailing ".psf.gz", just the font name below.
-# To use the default console font, comment out the CONSOLEFONT setting below.
-consolefont="default8x16"
+consolefont="default8x16.psf.gz"
 
 # consoletranslation is the charset map file to use.  Leave commented to use
 # the default one.  Have a look in /usr/share/consoletrans for a selection of
 # map files you can use.
-#consoletranslation="8859-1_to_uni"
-
-# unicodemap is the unicode map file to use. Leave commented to use the
-# default one. Have a look in /usr/share/unimaps for a selection of map files
-# you can use.
-#unicodemap="iso01"
+#consoletranslation="8859-1_to_uni.trans"
diff --git a/init.d/consolefont.in b/init.d/consolefont.in
index 4737516d..c51d7435 100644
--- a/init.d/consolefont.in
+++ b/init.d/consolefont.in
@@ -22,7 +22,6 @@ start()
 {
 	ttyn=${rc_tty_number:-${RC_TTY_NUMBER:-12}}
 	consolefont=${consolefont:-${CONSOLEFONT}}
-	unicodemap=${unicodemap:-${UNICODEMAP}}
 	consoletranslation=${consoletranslation:-${CONSOLETRANSLATION}}
 
 	if [ -z "$consolefont" ]; then
@@ -43,9 +42,6 @@ start()
 	if [ -n "$consoletranslation" ]; then
 		param="$param -m $consoletranslation"
 	fi
-	if [ -n "${unicodemap}" ]; then
-		param="$param -u $unicodemap"
-	fi
 
 	# Set the console font
 	ebegin "Setting console font [$consolefont]"
@@ -63,7 +59,8 @@ start()
 	# Store the font so we can use it ASAP on boot
 	if [ $retval -eq 0 ] && checkpath -m 0755 -d "$RC_CACHEDIR" 2>/dev/null; then
 		mkdir -p "$RC_CACHEDIR"/console
-		setfont -O "$RC_CACHEDIR"/console/font
+		zcat "/usr/share/consolefonts/$consolefont" \
+			> "$RC_CACHEDIR"/console/font
 	fi
 
 	return $retval
